src_c/transform.c

```diff
@@ -627,8 +627,8 @@ surf_scale_by(PyObject *self, PyObject *args, PyObject *kwargs)
     surf = pgSurface_AsSurface(surfobj);
     if (!surf)
         return RAISE(pgExc_SDLError, "display Surface quit");
-    newsurf = scale_to(surfobj, surfobj2, (int)(surf->w * scalex),
-                       (int)(surf->h * scaley));
+    newsurf = scale_to(surfobj, surfobj2, round(surf->w * scalex),
```

Comment:
Hey, maybe the build is failing, because this is C and not Python?
This should be `(int)(surf->w * scalex + 0.5)` and same for `scaley` below. This will do rounding instead of truncating.

Reply:
Or maybe not? I hear C99 has `round()`. Not sure how it is across compilers (like on MacOS...)